File: fluentcore/src/main/java/com/mcreater/fxfluent/brush/SolidColorBrush.java
package com.mcreater.fxfluent.brush;

import javafx.scene.layout.BorderWidths;
import javafx.scene.paint.Color;

public class SolidColorBrush extends AbstractColorBrush {
    private final Color color;
    public SolidColorBrush(Color color) {
        this.color = color != null ? color : Color.TRANSPARENT;
    }

    public String toString() {
        return getPaint().toString();
    }

    public Color getPaint() {
        return color;
    }

    public BorderWidths getBorderWidths() {
        return new BorderWidths(1);
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/brush/SolidColorBrush.kt
package com.mcreater.fxfluent.brush

import javafx.scene.paint.Color


class SolidColorBrush(color: Color?) : AbstractColorBrush() {
    private val color: Color

    init {
        this.color = color ?: Color.TRANSPARENT
    }

    override fun toString(): String {
        return getPaint().toString()
    }

    override fun getPaint(): Color? {
        return color
    }
}

